gsa/scripts/gsa_prepare_data.sh

```diff
@@ -1,40 +1,43 @@
 #!/bin/bash    
 
+[ -z ${ATLAS_EXPS+x} ] && echo "Env var ATLAS_EXPS for the path to atlas experiments needs to be defined." && exit 1
+[ -z ${ATLAS_GSA_FTP_DIR+x} ] && echo "Env var ATLAS_GSA_FTP_DIR for the path to the atlas ftp gsa folder needs to be defined." && exit 1
+
+WORKDIR="${ATLAS_GSA_WORKDIR:-${TMPDIR}/gsa}"
+
 # This script creates database files (one per organism in Atlas) to be used in the on-the-fly gene set enrichment against differentially expressed genes in each comparisons in a given Atlas organism
 
 scriptDir=$(cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
-source ${scriptDir}/../../../bash_util/generic_routines.sh
 
 # quit if not prod user
-check_prod_user
+atlas-bash-util check_prod_user
 if [ $? -ne 0 ]; then
     exit 1
 fi
 
-pushd $ATLAS_EXPS
-wDir=~/tmp/gsa
-mkdir -p $wDir
-rm -rf $wDir/*
-gsaFTPDir=/ebi/ftp/pub/databases/arrayexpress/data/atlas/gsa/
+pushd $ATLAS_EXPS > /dev/null
+mkdir -p $WORKDIR
+rm -rf $WORKDIR/*
 
 echo "About to assemble organism-experiments tuples"
-grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${wDir}/exps.aux
-awk '$1 = tolower($1)' ${wDir}/exps.aux > ${wDir}/exps.aux.lc && mv ${wDir}/exps.aux.lc ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
-perl -pi -e 's|^oryza_sativa_japonica |oryza_sativa_japonica_group |g' ${wDir}/exps.aux
+grep organism contrastdetails.tsv | grep -P '\torganism\t' | awk -F"\t" '{print $6"\t"$1}' | sort | uniq | sed 's| |_|g' > ${WORKDIR}/exps.aux
+awk '$1 = tolower($1)' ${WORKDIR}/exps.aux > ${WORKDIR}/exps.aux.lc && mv ${WORKDIR}/exps.aux.lc ${WORKDIR}/exps.aux
+perl -pi -e 's|^oryza_sativa |oryza_sativa_japonica_group |g' ${WORKDIR}/exps.aux
```

Comment:
We should probably leave an open issue here on these dangling species transformations. We took a more centralised approach in bulk recalculations based on a shared YAML file that describe these mappings. Not sure if those directly apply here, but we should keep track maybe with an issue that this is here.

Reply:
Yes, I do the same thing in single-cell. We should centralise that config as part of https://github.com/ebi-gene-expression-group/atlas-prod/pull/229

Reply:
Will make the issue